kube-client/src/config/file_config.rs

```diff
@@ -18,7 +19,7 @@ use super::{KubeconfigError, LoadDataError};
 ///
 /// [`Config`][crate::Config] is the __intended__ developer interface to help create a [`Client`][crate::Client],
 /// and this will handle the difference between in-cluster deployment and local development.
-#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+#[derive(Clone, Debug, Serialize, Deserialize, Default, PartialEq)]
```

Comment:
I think these can be `Eq` as well.

Reply:
I'm not sure how useful it actually is to expose `Eq` or `PartialEq` here tbh. People would run into weird cases pretty quickly, like expiry tokens being different or missing, even though the config is funcitonally the same.

Presonally, I was thinking we could maybe just limit the eq to tests with: `#[cfg_attr(test, derive(PartialEq))]`, and not expose this at all. Users probably need to think about this.

Reply:
Sure, will add this shortly

Reply:
Done

Reply:
Ah, I am not sure we have reached agreement on this one yet. Unresolving.

Reply:
Sorry, missed your comments - the browser didn't refresh the page properly.
So what do you guys think? I am fine with reverting back to multiple asserts in tests and dropping `PartialEq` if it does more harm than good.
While at that - I had a couple of situations where I needed `Kubeconfig` to be `Hash` - any reason not to?
In any case, I am happy to explore it in a different PR if it is not trivial

Reply:
Np. I would have to see a bit of an argument for `Hash`. As a baseline, personally, I am inclined to leave things like `Eq`, `PartialEq`, `Hash` out because of the potential footguns they can be (suddenly multiple equivalent configs flying around just because they have different `preferences.colors` or refresh tokens expiring at different times (say)). I think it would be safer for the user to define what they mean by equivalence - if they need to sort/order/compare/use as keys in a hashmap - by implementing the required traits themselves on a newtype. I'm expecting all of the use cases for these to be edge cases, but maybe I am wrong.

Curious to see what others think.

Reply:
Alright, I'll pursue `Hash` in a separate PR then. As for `PartialEq` and `Eq` I guess it is no, right?

Reply:
So, I've dropped it. And since the unit test now checks for failure rather than success it is irrelevant anyhow.

Reply:
Given the decision on the below issue, keep deriving PartialEq, but keep it behind cfg_attr test:

```
#[cfg_attr(test, derive(PartialEq))]
```

Reply:
Perhaps adding `Kubeconfig::is_empty` may be better in this case? Limiting `PartialEq` for tests only feels like special casing and error prone. Dedicated method can be more precise. WDYT?

Reply:
I'd rather not introduce extra helpers at this stage. It's not clear to me what `is_empty` would mean here because it's not clear what people mean by equivalence (why we don't want to export PartialEq in the first place).

I don't think limiting `PartialEq` for tests is more error prone than anything. The compiler ensures it's derived either no-where (public api) or everywhere (tests) because of how the instructions have to be propagated. It's an internal property, so this is a safe way to limit the public api.

Reply:
Alright